src/data.rs

```diff
@@ -450,6 +453,99 @@ pub fn decode_block_cid_ask_message(msg: Vec<u8>) -> Option<(i128, Option<Cid>)>
 	}
 }
 
+pub async fn ipfs_priority_get_cells(
+	cells: &mut Vec<DataCell>,
+	ipfs: &Ipfs<DefaultParams>,
+	rpc_url: &String,
+	block_num: u64,
+) -> Result<()> {
+	println!("ipfs priority get started...");
+	// Try fetching the cells from IPFS first
+	for (_, req_cell_data) in cells.iter_mut().enumerate() {
+		let peers = ipfs.peers();
+		println!("number of peers: {}", peers.len());
+		if peers.len() == 0 {
+			break;
+		}
+		// println!("list of connections: {:?}", ipfs.connections());
+		// TODO: paralelize fetching
+		// Skip cells that return error when fetching from IPFS
+		println!(
+			"Requesting from IPFS: CID: {}. Cell reference: {}.",
+			req_cell_data.cid(),
+			req_cell_data.reference()
+		);
+		let mut block_cid: Cid = Cid::default();
+		if let Ok(record) = ipfs
+			.get_record(
+				Key::from(req_cell_data.reference().as_bytes().to_vec()),
+				ipfs_embed::Quorum::One,
+			)
+			.await
+		{
+			block_cid = Cid::try_from(record[0].record.value.to_vec()).unwrap();
+		}
+
+		if let Err(error) = ipfs.fetch(&block_cid, peers).await.and_then(|block| {
+			// IPFS block data contains cell proof
+			println!("BLOCK DATA: {:?}", block.data());
+			req_cell_data.proof = block.data().to_vec();
+			req_cell_data.data = req_cell_data.proof[48..].to_vec();
```

Comment:
Is it ok to have proof with data (80 bytes instead of 48)?

Reply:
The `verify_proof` function takes the 80 bytes proof, then splits it as needed. For now, we're coupled to that implementation...